hw/xbox/nv2a/nv2a_vsh.c

```diff
@@ -674,13 +674,25 @@ static const char* vsh_header =
     "#define EXP(dest, mask, src) dest.mask = _EXP(_in(src).x).mask\n"
     "vec4 _EXP(float src)\n"
     "{\n"
-    "  return vec4(exp2(src));\n"
+    "  vec4 result;\n"
+    "  result.x = exp2(floor(src));\n"
+    "  result.y = src - floor(src);\n"
```

Comment:
I've mentioned this on IRC before; from 2.14.1.11 (Vertex Program Floating Point Requirements):

> The floor operations used by the ARL and EXP instructions must
> operate identically.  Specifically, the EXP instruction's floor(t.x)
> intermediate result must exactly match the integer stored in the
> address register by the ARL instruction.

Also see https://github.com/xqemu/xqemu/pull/79 for our current hack, which breaks this assumption.
Although I can't think of an example algorithm, having these in sync could be very critical, so we should respect this (there's probably good a reason why the spec is so specific).

I suggest adding our own `biased_floor` function or global constant for the bias (maybe pick a better name which respects our GLSL style), and using it in ARL and EXP.

Please also think about cases where our bias might run into problems (I worry about the result.y line because it might break assumption about returning [0.0;1.0[ ), so we can review the use of a bias or change it's value to avoid those problems.
So if it would break assumptions, we might have to add the bias to `src` for *all* operations in EXP.

I'm also fine with just documenting this as an issue if we can't find a good solution to integrate the bias.